src/components/Game.js

```diff
@@ -1,39 +1,79 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import Analyzer from './Analyzer.js';
+import Notes from './NoteDetails.js';
+import Info from './Info.js';
 
 class Game extends Component {
   constructor(props){
     super(props);
     this.state={
-      // last_propmted_note: null,
-      // current_prompted_note: null,
-      currentUserNote: null,
+      lastPromptedNoteNum: null,
+      promptedNoteNum: null,
+      promptedNoteLetter: null,
+      promptedNoteFreq: null,
+      userPlayingNote: null,
       // note_matched: false
+      infoShown: false
     }
+  this.toggleInfoShown = this.toggleInfoShown.bind(this);
+  this.increaseSkippedCountCallback = props.increaseSkippedCountCallback.bind(this);
+  }
 
+  toggleInfoShown () {
+    this.setState({ infoShown: !this.state.infoShown });
   }
 
-  getCurrentUserNote = (note) => {
-    this.setState({currentUserNote: note})
-    console.log('in game, currentUserNote: ', this.state.currentUserNote);
+  componentDidMount() {
+    this.setNewNote();
   }
 
-  componentWillUnmount() {
-    if (this.state.processor){this.state.processor.disconnect()};
+  setNewNote() {
+    let noteNum = null;
+    let noteDetails = null;
+    while(noteNum === null || noteDetails === undefined || noteNum === this.state.lastPromptedNoteNum) {
```

Comment:
this conditional seems to complex and error prone, noteDetails will be `undefined` when the random number matches one of the commented out values, why even allow for that to happen? should the notes be an array instead of an object? and the random number can be the index?

```
let nextNote;
do {
    const index = this.getRandomIntInclusive(0, Notes .length - 1);
    nextNote = Notes[index];
} while (nextNote === this.state.lastPrompedNote)
```

Reply:
Let's talk about this tomorrow. It goes along with my comment above related to using numbers as keys.